applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -133,18 +132,23 @@ class GeometryUtilities
     ///@{
 
     // ==============================================================================
+    void setPrecisionForOutput()
+    {
+        std::cout.precision(12);
+    }
 
+    // --------------------------------------------------------------------------
     void compute_unit_surface_normals()
     {
         KRATOS_TRY;
 
         // Compute nodal are normal using given Kratos utilities (sets the variable "NORMAL")
         NormalCalculationUtils normal_util = NormalCalculationUtils();
-        const unsigned int domain_size = mr_model_part.GetProcessInfo().GetValue(DOMAIN_SIZE);
-        normal_util.CalculateOnSimplex(mr_model_part,domain_size);
+        const unsigned int domain_size = mrModelPart.GetProcessInfo().GetValue(DOMAIN_SIZE);
+        normal_util.CalculateOnSimplex(mrModelPart,domain_size);
 
         // Take into account boundary conditions, normalize area normal and store in respective variable
-        for (ModelPart::NodeIterator node_i = mr_model_part.NodesBegin(); node_i != mr_model_part.NodesEnd(); ++node_i)
+        for (ModelPart::NodeIterator node_i = mrModelPart.NodesBegin(); node_i != mrModelPart.NodesEnd(); ++node_i)
         {
             // Normalize normal and assign to solution step value
             array_3d area_normal = node_i->FastGetSolutionStepValue(NORMAL);
```

Comment:
if i was in you i would write the following as

      array_1d<double,3>& nnormalized = node_i->FastGetSolutionStepValue(NORMALIZED_SURFACE_NORMAL);
      const array_1d<double,3>& n = node_i->FastGetSolutionStepValue(NORMAL);
      noalias(nnormalized) = n/norm_2(n);

this way you avoid two memory allocations.